src/main/java/org/openstreetmap/atlas/checks/validation/areas/OverlappingAOIPolygonCheck.java

```diff
@@ -0,0 +1,237 @@
+package org.openstreetmap.atlas.checks.validation.areas;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.clipping.Clip;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+import com.vividsolutions.jts.geom.TopologyException;
+
+/**
+ * This check flags {@link Area}s that overlap and represent the same Area of Interest (AOI) AOIs
+ * are defined by {@link TaggableFilter}s, which are grouped in a {@link List}.
+ *
+ * @author danielbaah
+ * @author bbreithaupt
+ */
+public class OverlappingAOIPolygonCheck extends BaseCheck
```

Comment:
Really high-level question - are there AOIs represented by Relations? If so, we want some kind of ComplexEntity implementation to walk the relation and build a valid Polygon that represents the AOI and update this check for completeness. If this is the case, can you please add an issue to the Atlas project for such functionality? Or you can confirm here, and I can make the issue :)

Reply:
I am not sure. I will check and report back.

Reply:
Cool - let me know what you find, might be an edge case.

Reply:
It does appear that AOIs are commonly represented by relations. If you are still up for putting in that Atlas issue, I think it might be best for you to do it as I am not particularly familiar with ComplexEntity.

Reply:
Sure, I can create the issue and we can update this check when it's ready (not in the near future). Can you provide some of the examples you found?

Reply:
A relations representing an AOI:
https://www.openstreetmap.org/relation/2985897

A relation representing an AOI overlapping a way representing an AOI:
https://www.openstreetmap.org/relation/1279007
https://www.openstreetmap.org/way/583819702

Reply:
Thanks! These are great examples. I've created the issue here: https://github.com/osmlab/atlas/issues/163. Once that's implemented, we can come back and update this check to capture more use cases.